magicctapipe/io/io.py

```diff
@@ -66,6 +58,55 @@
 DEAD_TIME_LST = 7.6 * u.us
 DEAD_TIME_MAGIC = 26 * u.us
 
+def telescope_combinations(config):
+    """
+    Generates all possible telescope combinations without repetition. E.g.: "LST1_M1", "LST2_LST4_M2", "LST1_LST2_LST3_M1" and so on.
+
+    Parameters
+    ----------
+    config: dict
+        yaml file with information about the telescope IDs. Typically evoked from "config_general.yaml" in the main scripts.
+
+    Returns
+    -------
+    TEL_NAMES: dict
+        Dictionary with telescope IDs and names.
+    TEL_COMBINATIONS: dict
+        Dictionary with all telescope combinations with no repetions.
+    """
+    
+    
+    TEL_NAMES = {}
+    for k, v in config["mc_tel_ids"].items():    #Here we swap the dictionary keys and values just for convenience.
+        if v > 0:
+            TEL_NAMES[v] =  k
+    
+    TEL_COMBINATIONS = {}
+    keys = list(TEL_NAMES.keys())
+    
+    def recursive_solution(current_tel, current_comb):
+    
+        if current_tel == len(keys):     #The function stops once we reach the last telescope
```

Comment:
This line means that the telescope ids in the config file need to be 1,2,...,n without any number missing right? And also in the right order?
If so I am wondering if there is a way to make it a bit more flexible.

Reply:
Actually I think it works also for arbitrary numbering: current_tel is not the telescope number but the index position of the telescope in the dictionary

Reply:
Ok good

Reply:
I just did the test using different telescope IDs or even deleting some lines, and everything worked perfectly. For instance, if we consider the following (unusual) configuration:
mc_tel_ids:
    LST-1: 2
    LST-4: 0
    MAGIC-I: 1
    MAGIC-II: 4

This function gives us the following output:
TEL_NAMES = {2: 'LST-1', 1: 'MAGIC-I', 4: 'MAGIC-II'}
TEL_COMBINATIONS = {'LST-1_MAGIC-I': [2, 1], 'LST-1_MAGIC-I_MAGIC-II': [2, 1, 4], 'LST-1_MAGIC-II': [2, 4], 'MAGIC-I_MAGIC-II': [1, 4]}